cassandra/policies.py

```diff
@@ -999,6 +999,237 @@ def shutdown(self):
             self.is_shutdown = True
 
 
+class ShardConnectionBackoffSchedule(ABC):
+    @abstractmethod
+    def new_schedule(self) -> Iterator[float]:
+        """
+        This should return a finite or infinite iterable of delays (each as a
+        floating point number of seconds).
+        Note that if the iterable is finite, schedule will be recreated right after iterable is exhausted.
+        """
+        raise NotImplementedError()
+
+
+class ConstantShardConnectionBackoffSchedule(ShardConnectionBackoffSchedule):
+    """
+    A :class:`.ShardConnectionBackoffSchedule` subclass which introduce a constant delay with jitter
+     between shard connections.
+    """
+
+    def __init__(self, delay: float, jitter: float = 0.0):
+        """
+        `delay` should be a floating point number of seconds to wait in-between
+        each connection attempt.
+
+        `jitter` is a random jitter in seconds.
+        """
+        if delay < 0:
+            raise ValueError("delay must not be negative")
+        if jitter < 0:
+            raise ValueError("jitter must not be negative")
+
+        self.delay = delay
+        self.jitter = jitter
+
+    def new_schedule(self):
+        if self.jitter == 0:
+            yield from repeat(self.delay)
+        def iterator():
+            while True:
+                yield self.delay + random.uniform(0.0, self.jitter)
+        return iterator()
+
+
+class LimitedConcurrencyShardConnectionBackoffPolicy(ShardConnectionBackoffPolicy):
+    """
+    A shard connection backoff policy that allows only `max_concurrent` concurrent connections per `host_id`.
+
+    For backoff calculation, it requires either a `cassandra.policies.ShardConnectionBackoffSchedule` or
+    a `cassandra.policies.ReconnectionPolicy`, as both expose the same API.
+
+    It spawns threads when there are pending requests, maximum number of threads is `max_concurrent` multiplied by nodes in the cluster.
+    When thread is spawn it initiates backoff schedule, which is local for this thread.
+    If there are no remaining requests for that `host_id`, thread is killed.
+
+    This policy also prevents multiple pending or scheduled connections for the same (host, shard) pair;
+    any duplicate attempts to schedule a connection are silently ignored.
+    """
```

Comment:
So this comment is saying about concurrent connections, and spawning threads. As far as I can tell, none of those things are happening here. 
Scheduler we are using here for opening connections has 1 thread, so there is no concurrency happening.
The class does not spawn threads anywhere, so idk where this comment comes from.

Reply:
Got it, it is a bit confusing, changed thread to worker, add more information to `_ScopeBucket`
`_Scheduler` has one thread, but it does not run scheduled code, it uses `cluster.executor` for that, which has 2 threads.